### PR TITLE
[ACR] dedicated data endpoint

### PR DESCRIPTION
**Pull request by yugangw-msft**
**Description**  
Onboard dedicated data endpoint support

**Testing Guide**  
"az acr update -n myreg --data-endpoint-enabled"
"az acr show-endpoints"

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S169

**Comment by yugangw-msft** [comment]
> one minor question will it be appropriate list-endpoints

Because the returning type is an object (with endpoints info inside), instead of a list, so we use show. Let me talk to @SteveLasker for whether we have a better naming

**Comment by yungezz** [comment]
> > one minor question will it be appropriate list-endpoints
> 
> Because the returning type is an object (with endpoints info inside), instead of a list, so we use show. Let me talk to @SteveLasker for whether we can have a better naming

make sense, thanks @yugangw-msft.  approved already.

File: src/azure-cli/azure/cli/command_modules/acr/_help.py [hunk 1]
```diff
@@ -1288,4 +1288,9 @@
 type: command
 short-summary: Show the container registry's identity details
 """
+
+helps['acr show-endpoints'] = """
+type: command
+short-summary: display the list of endpoints
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acr/_help.py:1294]:
```suggestion
short-summary: Display the list of endpoints
```

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 2]
```diff
@@ -84,6 +84,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('identity', help="Use assigned managed identity resource id or name if in the same resource group")
         c.argument('key_encryption_key', help="key vault key uri")
 
+    with self.argument_context('acr update', is_preview=True) as c:
+        c.argument('data_endpoint_enabled', get_three_state_flag(), help="enable dedicated data endpoint for client firewall configuration")
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acr/_params.py:88]:
```suggestion
        c.argument('data_endpoint_enabled', get_three_state_flag(), help="Enable dedicated data endpoint for client firewall configuration")
```

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:88]: CLI will format help automatically. But it is a good practice to write sentence with correct grammar.

**Comment by yugangw-msft** [comment]
I have addressed all review comments. Please merge once CI is green.

**Pull request merged**